model/src/main/java/io/syndesis/model/connection/ActionMetadata.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.model.connection;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import io.syndesis.model.ImmutablesStyle;
+import io.syndesis.model.WithName;
+import io.syndesis.model.WithProperties;
+
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonDeserialize(builder = ActionMetadata.Builder.class)
+public interface ActionMetadata {
+
+    final class Builder extends ImmutableActionMetadata.Builder {
+        // make ImmutableActionMetadata.Builder accessible
+
+        public Builder withStep(final String name, final String description, final Consumer<Step.Builder> stepConfigurator) {
+            final Step.Builder stepBuilder = new Step.Builder().name(name).description(description);
+
+            stepConfigurator.accept(stepBuilder);
+
+            addPropertyDefinitionStep(stepBuilder.build());
+
+            return this;
+        }
+    }
+
+    @Value.Immutable
+    @JsonDeserialize(builder = Step.Builder.class)
+    @ImmutablesStyle
+    interface Step extends WithName, WithProperties {
```

Comment:
We already have `Step` as part of an integration definition. Maybe we can call it differently here like in `ActionDefinitionStep` ?